### PR TITLE
feat: mobile ui svg icons

### PR DESCRIPTION
**Pull request by fbwoolf**
This PR relocates all svgs from the mobile app into the UI library. They are now exported from the UI library as React components with a default size of 24x24 and a 'small' variant of 16x16. All other SvgProps can be passed to change color, size, etc. The theme is used in the UI library to set the 'currentColor' as 'ink.action-primary-default', so the icons work with dark mode.

The native UI library build is now using tsup (aka 'sup'). 😆 Once this is merged, I can refactor the existing 'web' icons to use the svgs, but I wanted to separate the work into two PRs.

Preview of use...
Screenshot 2024-08-15 at 4 21 42 PM

signal-2024-08-18-192320_002

signal-2024-08-18-192320_003

### CODE REVIEW

File: packages/ui/app.json [hunk 1]
```diff
@@ -31,6 +31,17 @@
...
-    "plugins": ["expo-font"]
+    "plugins": [
+      "expo-font",
+      {
+        "fonts": [
+          "../assets-native/fonts/SpaceMono-Regular.ttf",
+          "../assets-native/fonts/ABCDiatype-Regular.otf",
+          "../assets-native/fonts/ABCDiatype-Light.otf",
+          "../assets-native/fonts/ABCDiatype-Medium.otf",
+          "../assets-native/fonts/marche-super-pro.otf"
```
Comment (kyranjamie) [inline packages/ui/app.json:42]: Can we rename to be all the same casing?

File: packages/ui/package.json [hunk 2]
```diff
@@ -4,12 +4,12 @@
   "license": "MIT",
   "type": "module",
   "scripts": {
-    "assets:native": "mkdir -p dist-native/src && cp -r src/assets-native ./dist-native/src/",
-    "assets:web": "mkdir -p dist-web/src && cp -r src/assets-web ./dist-web/src",
+    "assets:native": "mkdir -p dist-native/src && cp -r src/assets-native ./dist-native/src && cp -r src/assets ./dist-native/src",
+    "assets:web": "mkdir -p dist-web/src && cp -r src/assets-web ./dist-web/src && cp -r src/assets ./dist-web/src",
```
Comment (kyranjamie) [inline packages/ui/package.json:8]: Not a fan of having these copy commands in the package.json. tsup doesn't have a "beforeBuild" like it does `onSuccess` but [per comment here](https://github.com/egoist/tsup/issues/1094#issuecomment-1987318528), anything in the build script runs before the config.

So, could we move these directory/asset copying actions there, using a node script to do the same?

**Review by kyranjamie — APPROVED** [review]
Fantastic work Fara, this is a really tidy set up. Definitely worth the time you spent investigating this to get to this result.

**Review by pete-watters — APPROVED** [review]
Great work @fbwoolf 👏

File: packages/ui/src/icons/arrow-left-icon.native.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      <Icon {...props}>
+        <ArrowLeftSmall />
+      </Icon>
+    );
+  return (
+    <Icon {...props}>
+      <ArrowLeft />
+    </Icon>
+  );
+}
```
[the remark this answers is not in the document]
Comment (kyranjamie) [reply packages/ui/src/icons/arrow-left-icon.native.tsx:17]: @edgarkhanzadian there are separate SVG files to handle small pixel rendering differences at smaller sizes

File: packages/ui/package.json (hunk 2 above)
Comment (fbwoolf) [reply packages/ui/package.json:8]: I meant to come back to this bc came across a suggestion to use this plugin to copy assets in the tsup config:
https://github.com/LinbuduLab/esbuild-plugins/tree/main/packages/esbuild-plugin-copy

File: packages/ui/app.json (hunk 1 above)
Comment (fbwoolf) [reply packages/ui/app.json:42]: Let me check this.

File: packages/ui/src/icons/arrow-left-icon.native.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (fbwoolf) [reply packages/ui/src/icons/arrow-left-icon.native.tsx:17]: @edgarkhanzadian the designers asked us not to scale svgs b/w 16x16 and 24x24, so we have two separate exports for the 'default' and 'small' variants. If another size is needed, we can then scale from the default.

**Comment by fbwoolf** [comment]
@kyranjamie I'm getting a 'not-to-dev' dependency error here with tsup. Docs say it should be installed as a dev dependency, but we had it as a dependency. I moved it and added `tsup` to the `pathNot`, let me know if that isn't the correct way to handle that error. I also gitignored the generated `.tsup` folder.

**Comment by kyranjamie** [comment]
Config files like tsup shouldn't be subject to rules of depcruiser, so good to ignore (even globally)

File: packages/ui/app.json (hunk 1 above)
Comment (fbwoolf) [reply packages/ui/app.json:42]: The fonts aren't broken that I can see, do we need them in both places?

Comment (fbwoolf) [reply packages/ui/app.json:42]: Nvm, think I am wrong here.

**Pull request merged**